infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/EphemeralWorkspaceAdapter.java

```diff
@@ -44,11 +50,59 @@
 
   private final PVCProvisioner pvcProvisioner;
   private final SubPathPrefixes subPathPrefixes;
+  private final boolean defaultPersistVolumes;
 
   @Inject
-  public EphemeralWorkspaceAdapter(PVCProvisioner pvcProvisioner, SubPathPrefixes subPathPrefixes) {
+  public EphemeralWorkspaceAdapter(
+      PVCProvisioner pvcProvisioner,
+      SubPathPrefixes subPathPrefixes,
+      @Named(CHE_WORKSPACE_PERSIST_VOLUMES_PROPERTY) boolean defaultPersistVolumes) {
     this.pvcProvisioner = pvcProvisioner;
     this.subPathPrefixes = subPathPrefixes;
+    this.defaultPersistVolumes = defaultPersistVolumes;
+  }
+
+  /**
+   * @param workspaceAttributes workspace config or devfile attributes to check is ephemeral mode is
+   *     enabled
+   * @return true if `persistVolumes` attribute exists and set to 'false'. In this case regardless
+   *     of the PVC strategy, workspace volumes would be created as `emptyDir`. When a workspace Pod
+   *     is removed for any reason, the data in the `emptyDir` volume is deleted forever
+   */
+  public boolean isEphemeral(Map<String, String> workspaceAttributes) {
+    String persistVolumes = workspaceAttributes.get(PERSIST_VOLUMES_ATTRIBUTE);
+    if (persistVolumes == null) {
+      return !defaultPersistVolumes;
```

Comment:
I'm not sure this is the logic we want to go with -- I'd propose
- If `persistVolumes` is not present, assumed default value is true.
- Only use ephemeral mode when `persistVolumes` is explicitly false.

This avoids the need to set it explicitly in all devfiles, and avoids breaking existing workspaces by disconnecting them from existing PVCs. 

On the workspace creation side (e.g. in the dashboard), this would be handled by explicitly setting `persistVolumes: false` if necessary, and leaving it out of the devfile otherwise.

Reply:
Put another way: if 
- `che.workspace.persist_volumes.default=false`, any workspaces created will additionally have `persistVolumes: false` in their devfiles upon creation. 
- `che.workspace.persist_volumes.default=true`, workspaces don't get an added `persistVolumes` field.

Reply:
I wanted to give an ability for Che Admin to configure default behavior of persisting volumes...
AFAIU you propose to keep the current Che Server logic where explicit value is used or volumes are persisted by default and only propagate default value to be consumed by clients like Dashboard.
I agree that it's safer in the perspective of existing workspaces without explicit values, where users do not expect changing behavior.
On another hand, Che Admin configured value won't be set to all workspaces by default, even new if workspaces are created with another client but not Dashboard - like chectl, or factory(if the value is not set explicitly in devfile).
I'm OK with the proposed alternative way since it's a lighter way that does not change the current Che Server behavior and at the same time allows us to configure Dashboard behavior via Che Server configuration.